File: flutter/library/lib/src/tsp_badge.dart
import 'package:flutter/material.dart';
import 'star_planet_theme.dart';

class TspBadge extends StatelessWidget {
  const TspBadge({super.key, required this.text, this.variant = 'default', this.disabled = false, this.theme = StarPlanetTheme.sky});
  final String text;
  final String variant;
  final bool disabled;
  final StarPlanetTheme theme;

  @override
  Widget build(BuildContext context) {
    final fill = switch (variant) {
      'primary' => theme.brandPrimary,
      'success' => theme.success,
      'warning' => theme.warning,
      'danger' => theme.danger,
      _ => theme.pageEnd,
    };
    final color = ['primary', 'success', 'danger'].contains(variant) ? Colors.white : theme.textPrimary;
    return Opacity(
      opacity: disabled ? .45 : 1,
      child: Container(
        constraints: const BoxConstraints(minHeight: 28),
        padding: const EdgeInsets.symmetric(horizontal: 10, vertical: 5),
        alignment: Alignment.center,
        decoration: ShapeDecoration(color: fill, shape: const StadiumBorder()),
        child: Text(text, style: TextStyle(color: color, fontWeight: FontWeight.w800, fontSize: 13, height: 16 / 13)),
      ),
    );
  }
}

File: ios-swiftui/library/Sources/PlanetComponents/TspBadge.swift
import SwiftUI

public struct TspBadge: View {
    let text: String
    let variant: String
    let theme: StarPlanetTheme

    public init(_ text: String, variant: String = "default", theme: StarPlanetTheme = .sky) {
        self.text = text
        self.variant = variant
        self.theme = theme
    }

    public var body: some View {
        Text(text)
            .font(.system(size: 13, weight: .bold))
            .foregroundColor(["primary", "success", "danger"].contains(variant) ? .white : theme.textPrimary)
            .padding(.horizontal, 10)
            .padding(.vertical, 6)
            .background(fill)
            .clipShape(Capsule())
    }

    private var fill: Color {
        switch variant {
        case "primary": return theme.brandPrimary
        case "success": return theme.success
        case "warning": return theme.warning
        case "danger": return theme.danger
        default: return theme.pageEnd
        }
    }
}
